lua/neoscroll/init.lua

```diff
@@ -35,7 +35,9 @@ end
 local function hide_cursor()
     if vim.o.termguicolors then
         guicursor = vim.o.guicursor
-        vim.o.guicursor = guicursor .. ',a:NeoscrollHiddenCursor'
+	if guicursor ~= '' then
```

Comment:
Let's move the condition `guicursor ~= ''` to line 36 as a second condition of the if statement. We don't need to store `guicursor` if we are not going to change it.